### PR TITLE
Fixed a small bug in metric wrapper

### PR DESCRIPTION
**Pull request by Seilmast**
Found that some metrics might not have made sure all metrics are on the same device. Made the MetricWrapper force `y_true` and `y_logits` onto the cpu

### CODE REVIEW

**Review by Johanmkr — APPROVED** [review]
Perfect!
